spec/system/candidate_interface/offers_and_withdrawals/candidate_accepts_offer_new_reference_feature_enabled_spec.rb

```diff
@@ -0,0 +1,464 @@
+require 'rails_helper'
+
+RSpec.feature 'Candidate accepts an offer' do
```

Comment:
I think the size of this system spec is concerning in terms of maintenance, and probably suggests that too much path-varying logic is held in the controller and view tier as we step through every permeation of the flow.

Generally system specs would test the happy path and we sometimes write a secondary spec to test some flow logic.
As discussed, by having something which understands the various contextual paths given the controller, action and params we would have a central place to check that all these path aliasing methods are correct. This can then be unit tested with much fewer test dependencies, we could then reduce the size of this spec.

Reply:
I see what you mean. I agree, I think until we have the path resolver the implementation will contain code smells. We shouldn't postpone. The next card needs a similar logic so probably there will be time to tackle resolving paths and remove the view and controller bloating helper methods.